src/hooks/index.ts

```diff
@@ -59,21 +69,26 @@ export function useInactiveListener(suppress = false) {
                     activate(injected, undefined, true).catch(error => {
                         console.error('Failed to activate after accounts changed', error)
                     })
+                } else {
+                  deactivate() 
                 }
             }
+            
+            ethereum.selectedProvider.on('chainChanged', handleChainChanged) 
+            ethereum.selectedProvider.on('accountsChanged', handleAccountsChanged)
 
-            ethereum.on('chainChanged', handleChainChanged)
-            ethereum.on('accountsChanged', handleAccountsChanged)
 
             return () => {
+              // Not sure why, but the actual listener exists here
+              // .on has to be set through the selectedProvider to work??
                 if (ethereum.removeListener) {
                     ethereum.removeListener('chainChanged', handleChainChanged)
                     ethereum.removeListener('accountsChanged', handleAccountsChanged)
```

Comment:
selectedProvider?

Reply:
I have no idea as well!

This works for all three scenario's: Only Meta / Only CoinBase / Both

Any other setup and the event-listeners stop working, and I cannot figure out why
the scope inside the ethereum object gets (for me) really confusing with coinbase in it

Or should I need to add them both?